addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+        if (refreshLogin != null) {
+            refreshLogin.cancel(true);
+            refreshLogin = null;
+        }
+        if (connection != null) {
+            connection.logout();
+            connection = null;
+        }
+    }
+
+    private void start() {
+        logger.debug("amazon account bridge starting handler ...");
+
+        config = getConfigAs(AccountConfiguration.class);
+        if (config.amazonSite == null || config.amazonSite.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Amazon site not configured");
+            cleanup();
+            return;
+        }
+        if (config.email == null || config.email.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Account email not configured");
+            cleanup();
+            return;
+        }
+        if (config.password == null || config.password.isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Account password not configured");
+            cleanup();
+            return;
+        }
+        if (config.pollingIntervalInSeconds == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Polling interval not configured");
+            cleanup();
+            return;
+        }
+        if (config.discoverSmartHomeDevices != null && config.discoverSmartHomeDevices) {
+            if (!smartHodeDeviceListEnabled) {
+                updateSmartHomeDeviceList = true;
+            }
+            smartHodeDeviceListEnabled = true;
+        } else {
+            smartHodeDeviceListEnabled = false;
+        }
+
+        if (config.pollingIntervalInSeconds < 10) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval less than 10 seconds not allowed");
+            cleanup();
+            return;
+        }
+        synchronized (synchronizeConnection) {
+            if (connection == null || !connection.getEmail().equals(config.email)
+                    || !connection.getPassword().equals(config.password)
+                    || !connection.getAmazonSite().equals(config.amazonSite)) {
+                connection = new Connection(config.email, config.password, config.amazonSite,
+                        this.getThing().getUID().getId());
+            }
+        }
+        if (this.loginServlet == null) {
+            this.loginServlet = new LoginServlet(httpService, this.getThing().getUID().getId(), this, config);
+        }
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Wait for login");
+        if (refreshLogin != null) {
+            refreshLogin.cancel(false);
+        }
+        refreshLogin = scheduler.scheduleWithFixedDelay(() -> {
```

Comment:
use `this::checkLogin` instead, please apply where possible

Reply:
Sorry, but I don't understand this comment. Should I rename the scheduler variable refreshLogin? What does this syntax with :: mean? I'am new in Java, please help!

Reply:
Since java 8 you can use method references with eg. this::methodName  this instead of the () -> {

Reply:
OK
Thanks for clarification!